TypeScript/HW_12/task 12.1.2.ts

```diff
@@ -0,0 +1,88 @@
+// 2. Создайте интерфейс Person, абстрактный класс Employee, который реализует интерфейс Person, и конкретные классы Manager и Developer.
+//   - Интерфейс Person должен содержать:
+//       Стринговые поля: name, surname; поле number - experienceYears,
+//       Метод, возвращающий строку: getDetails().
+
+//   - Абстрактный класс Employee должен:
+//       Реализовывать интерфейс Person.
+//       Содержать защищенное поле: salary, не передающееся в конструктор (по дефолту 0)
+//       Содержать защищенный абстрактный метод: calculateSalary().,
+//         который считает зарплату и вызывается в конструкторе, и будет реализован в наследниках
+  
+//   - Конкретные классы Manager и Developer должны:
+//       Наследоваться от Employee.
+//       Класс менеджер должен на конструкторе получать поле prefered, которое может быть только 'scrum' или 'kanban'
+//       Класс девелопер должен на конструкторе получать поле programmingLanguage, который может быть 'js', 'ts', 'java', 'python'
+//       Метод calculateSalary должен для менеджера устанавливать зарплату по формуле: количество лет опыта * 500
+//       Метод calculateSalary должен для девелопера устанавливать зарплату по формуле: количество лет опыта * 1000
+//       Реализовывать метод getDetails(), который должен выводить полную информацию об объекте вида:
+//         'My name is Elena TSovna, I am software developer with 6 years of experience in TypeScript and 6000$ salary' (пример для девелопера)
+
+interface Person {
+    name: string;
+    surname: string;
+    experienceYears: number;
+    getDetails(): string;
+};
+
+abstract class Employee implements Person {
+    protected salary: number = 0;
+    constructor (
+        public name: string,
+        public surname: string,
+        public experienceYears: number
+    )
+    { 
+        this.getDetails();
```

Comment:
Why we need to use this method in constructor and not `calculateSalary` ?

Reply:
I read the terms of the problem again: I mixed up the methods by mistake. Fixed